### ISSUE TITLE
imported self-referencing generic type aliases unhandled

### ISSUE THREAD

**Issue by StarrFox** [microsoft/pyright#6664]
**Describe the bug**
imported generic type aliases that reference themselves cause an unhandled exception

should instead get an error like the one for non-generic type aliases that reference themselves

**Code or Screenshots**

a.py:
```py
from b import A, B
type A = A[B]
```
b.py:
```py
class A[T]: ...
class B: ...
```

then do `pyright a.py`

error: https://gist.github.com/StarrFox/0608c9a82fe9061a7c220b9e69d45d6f

**VS Code extension or command-line**
```
❯ pyright --version
pyright 1.1.335
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.340, which I just published. It will be included in a future release of pylance.
